### PR TITLE
auto generate buttons

### LINKED ISSUE #1098: auto-generate "new activity" and "new upgrade" buttons for activity grids
This has been marked as a good candidate for becoming a [first-timers-only issue](http://www.firsttimersonly.com) like [these](https://github.com/publiclab/plots2/labels/first-timers-only), meaning that it's simple, self-contained, and with some extra formatting, could be a great entry point for a new contributor. If you're familiar enough with this code, please consider [reformatting or reposting it as a first-timers-only issue](https://publiclab.org/notes/warren/10-31-2016/create-a-welcoming-first-timers-only-issue-to-invite-new-software-contributors), and then ping @jywarren to get it labelled. Or, if this is not your first time, try to solve it yourself!

****

Our great [new activity grids](https://publiclab.org/notes/liz/08-30-2016/check-out-these-activity-grids) display lists of activities based on tag name, and are often accompanied by buttons that say `Add an activity` and `Request an activity`. These are currently manually added, for example on this page:

https://publiclab.org/wiki/kite-making

Along with a helpful notice below, using this mix of HTML and markdown:

```
## Activities

[activities:kite-making]

<a href="/post?tags=activity:kite-making,kite-making,seeks:replications&title=How%20to%20do%20X" class="btn btn-primary">Add an activity</a> <a href="/post?tags=question:kite-making,request:activity&template=question&title=How%20do%20I...&redirect=question" class="btn btn-primary">Request an activity guide</a>

_Guides should include a materials list and a step-by-step construction guide with photo documentation. Learn what [makes a good activity here](https://publiclab.org/notes/warren/09-17-2016/what-makes-a-good-activity)._
```

(screenshot would be helpful here)

That's long and we could automate it by adding a new line to the template generator in this file, just after this line:

https://github.com/publiclab/plots2/blob/master/app/helpers/application_helper.rb#L55

It could just be the HTML from above, so, substituting in the `tagname`:

```ruby
      output += '<a href="/post?tags=activity:#{tagname},#{tagname},seeks:replications&title=How%20to%20do%20X" class="btn btn-primary add-activity">Add an activity</a> <a href="/post?tags=#{tagname},question:#{tagname},request:activity&template=question&title=How%20do%20I...&redirect=question" class="btn btn-primary request-activity">Request an activity guide</a>'
      output += '_Activities should include a materials list, costs and a step-by-step guide to construction with photos. Learn what [makes a good activity here](/notes/warren/09-17-2016/what-makes-a-good-activity)._'
```

### PR DESCRIPTION
**Pull request by mridulnagpal**
Make sure these boxes are checked before your pull request is ready to be reviewed and merged. Thanks!

* [x] all tests pass -- `rake test:all`
* [x] code is in uniquely-named feature branch, and has been rebased on top of latest master (especially if you've been asked to make additional changes)
* [x] pull request is descriptively named with #number reference back to original issue
* [x] if possible, multiple commits squashed if they're smaller changes
* [x] reviewed/confirmed/tested by another contributor or maintainer
* [x] `schema.rb.example` has been updated if any database migrations were added

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/wiki/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. Please alert developers on plots-dev@googlegroups.com when your request is ready or if you need assistance.
Closes #1098 
Thanks!

### CODE REVIEW

**Comment by mridulnagpal** [comment]
It gives an error 
<"Error: only the author may use that powertag."> expected but was
  <"Error: cannot find that username.">.
test_won't_add_disallowed_tags(TagControllerTest)
test/functional/tag_controller_test.rb:66:in `block in <class:TagControllerTest>'

**Comment by jywarren** [comment]
Hmm, are you rebased on top of most recent master?

**Comment by mridulnagpal** [comment]
Yeah just pulled from https://github.com/publiclab/plots2.git for master and created a new branch

**Comment by jywarren** [comment]
hmm sorry, i'll have to take a look later, that's odd!

**Comment by jywarren** [comment]
Oh look it passed!

**Comment by mridulnagpal** [comment]
@jywarren It looks like my remote must have had some issue I'll resolve it ASAP

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -53,6 +53,8 @@ def insert_extras(body)
         output += '</tr>'
       end
       output += '</table>'
+      output += '<a href="/post?tags=activity:#{tagname},#{tagname},seeks:replications&title=How%20to%20do%20X" class="btn btn-primary add-activity">Add an activity</a> <a href="/post?tags=#{tagname},question:#{tagname},request:activity&template=question&title=How%20do%20I...&redirect=question" class="btn btn-primary request-activity">Request an activity guide</a>'
```
Comment (jywarren) [inline app/helpers/application_helper.rb]: Can this be in a p tag?

**Comment by jywarren** [comment]
I made a small tweak, and will try to merge after tests rerun.

**Comment by jywarren** [comment]
Perfect, merged! Really getting them done tonight! @ebarry -- note that we'll publish this in the next couple days and will need button de-duplication once we do. I'll alert you, thanks!

**Comment by ebarry** [comment]
Great everyone! Please do alert me when these automatic/automagic :) buttons are live, and i'll go "wiki gardening".

**Comment by jywarren** [comment]
They're live I just knew you weren't working yet so held off on telling you!

**Pull request merged**